File: Exercism/dart/isogram/lib/isogram.dart

class Isogram {
  bool isIsogram(String input) {
    var letters = <String>{};
    var result = true;

    input.replaceAll(' ', '').replaceAll('-', '').toLowerCase().split('')
    .forEach((String l) {
        if (letters.contains(l))
          result = false;

        letters.add(l);
    });

    return result;
  }
}

File: Exercism/typescript/isogram/isogram.ts
class Isogram {
    static isIsogram(input: string) {
        let letters = new Set();

        for (var c of input.toLowerCase().split("")) {
            if (c == ' ' || c == '-')
                continue;
            if (letters.has(c))
                return false;
            letters.add(c);
        }

        return true;
    }
}

export default Isogram
